libs/portals/admin/ids-admin/src/components/forms/CreateApplication/CreateApplication.action.ts

```diff
@@ -0,0 +1,69 @@
+import { z } from 'zod'
+import { zfd } from 'zod-form-data'
+import { WrappedActionFn } from '@island.is/portals/core'
+import {
+  replaceParams,
+  validateFormData,
+  ValidateFormDataResult,
+} from '@island.is/react-spa/shared'
+import {
+  AuthAdminApplicationType,
+  AuthAdminEnvironment,
+} from '@island.is/api/schema'
+import {
+  CreateApplicationDocument,
+  CreateApplicationMutation,
+} from './CreateApplication.generated'
+import { redirect } from 'react-router-dom'
+import { IDSAdminPaths } from '../../../lib/paths'
+
+const schema = z.object({
+  displayName: z.string().nonempty('errorDisplayName'),
+  applicationId: z.string().nonempty('errorApplicationId'),
+  tenant: z.string(),
+  environments: zfd.repeatable(
+    z.array(z.nativeEnum(AuthAdminEnvironment)).nonempty('errorEnvironment'),
+  ),
+  applicationType: z.nativeEnum(AuthAdminApplicationType, {
+    required_error: 'errorApplicationType',
+  }),
+})
+
+export type CreateApplicationResult =
+  | ValidateFormDataResult<typeof schema>
+  | undefined
+
+export const createApplicationAction: WrappedActionFn = ({ client }) => async ({
+  request,
+}) => {
+  const result = await validateFormData({ request, schema })
+
+  if (result.errors || !result.data) {
+    return result
+  }
+
+  const { data } = result
+
+  const res = await client.mutate<CreateApplicationMutation>({
+    mutation: CreateApplicationDocument,
+    variables: {
+      input: {
+        displayName: data.displayName,
+        applicationId: `${data.tenant}/${data?.applicationId}`,
+        environments: data.environments,
+        applicationType: data?.applicationType,
+      },
+    },
+  })
+
+  if (res.errors) {
+    throw res.errors
```

Comment:
This will log in the console the error but React router error element will handle it and display the current route segment error or the parent one if not specified.

Reply:
~Nice to have would be to have some error box above the form and then we would have to handle this differently

Reply:
Now we have global error box